var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -57,5 +63,12 @@ class Mesa(AutotoolsPackage):
     depends_on('presentproto@1.0:', type='build')
     depends_on('pkg-config@0.9.0:', type='build')
 
+
+    def configure_args(self):
+        make('autoreconf -fi')
```

Comment:
> It looks like the newer version fixes this autoreconf problem (it looks like spack actually somehow calls autoreconf automatically)

Since #1186 sub-classes of `PackageBase` can declare an arbitrary number of phases to conform to the underlying build-system (`cmake`, `autotools`, etc.). The `AutotoolsPackage` class has logic in it to perform the most common operations (like call `autoreconf` if no `configure` file is present) so that packagers don't need to repeat them over and over.